### PR TITLE
Arm backend: Add numerically stable (log)softmax decomposition

### PR DESCRIPTION
**Pull request by AdrianLundell**
- Only use the old version for Ethos-U55 compile specs since amax isn't supported in that case.
- Add support for negative indices in amax/amin
- Refactor unittests

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218

### CODE REVIEW

**Comment by AdrianLundell** [comment]
Unrelated errors in [trunk / test-arm-reference-delegation / linux-job](https://hud.pytorch.org/pr/pytorch/executorch/9133#38624133556):
```
CMake Error at /pytorch/executorch/examples/arm/ethos-u-scratch/ethos-u/core_software/CMakeLists.txt:73 (add_subdirectory):
  The source directory

    /pytorch/executorch/examples/arm/ethos-u-scratch/ethos-u/core_software/core_driver

  does not contain a CMakeLists.txt file.
...
CMake Error at /pytorch/executorch/examples/arm/ethos-u-scratch/ethos-u/core_software/tflite_micro.cmake:134 (add_subdirectory):
  add_subdirectory given source
  "/pytorch/executorch/examples/arm/ethos-u-scratch/ethos-u/core_software/cmsis-nn"
  which is not an existing directory.
Call Stack (most recent call first):
  /pytorch/executorch/examples/arm/ethos-u-scratch/ethos-u/core_software/CMakeLists.txt:77 (include)
...
CMake Error at /pytorch/executorch/examples/arm/ethos-u-scratch/ethos-u/core_software/tflite_micro.cmake:136 (target_compile_options):
  Cannot specify compile options for target "cmsis-nn" which is not built by
  this project.
Call Stack (most recent call first):
  /pytorch/executorch/examples/arm/ethos-u-scratch/ethos-u/core_software/CMakeLists.txt:77 (include)
```

**Review by zingo — APPROVED** [review]
